passwordservice/service.go

```diff
@@ -0,0 +1,133 @@
+package passwordservice
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/lightningnetwork/lnd/lnrpc"
+	"github.com/lightningnetwork/lnd/lnwallet/btcwallet"
+	"github.com/roasbeef/btcd/chaincfg"
+	"github.com/roasbeef/btcd/wire"
+	"github.com/roasbeef/btcwallet/wallet"
+	"golang.org/x/net/context"
+)
+
+// PasswordService implements the WalletUnlocker service used to provide lnd
+// with a password for wallet encryption at startup.
+type PasswordService struct {
+	// CreatePasswords is a channel where passwords provided by the rpc
+	// client to be used to initially create and encrypt a wallet will
+	// be sent.
+	CreatePasswords chan []byte
+
+	// UnlockPasswords is a channel where passwords provided by the rpc
+	// client to be used to unlock and decrypt an existing wallet will
+	// be sent.
+	UnlockPasswords chan []byte
+
+	chainDir  string
+	netParams *chaincfg.Params
+}
+
+// New creates and returns a new PasswordService.
+func New(chainDir string, params *chaincfg.Params) *PasswordService {
+	return &PasswordService{
+		CreatePasswords: make(chan []byte, 1),
+		UnlockPasswords: make(chan []byte, 1),
+		chainDir:        chainDir,
+		netParams:       params,
+	}
+}
+func (p *PasswordService) CreateWallet(ctx context.Context,
+	in *lnrpc.CreateWalletRequest) (*lnrpc.CreateWalletResponse, error) {
+
+	cfg := &btcwallet.Config{
```

Comment:
What is the purpose of creating this `cfg`?

Reply:
removed